src/compiler/parser-decl-symbtype.bas

```diff
@@ -29,6 +29,85 @@ function cConstIntExpr _
 	function = astConstFlushToInt( expr, dtype )
 end function
 
+''
+type RANGEVALUES
+	as longint smin
+	as longint smax
+	as ulongint umax
+end type
+
+''
+function cConstIntExprRanged _
+	( _
+		byval expr as ASTNODE ptr, _
+		byval todtype as integer _
+	) as longint
+
+	static range( FB_SIZETYPE_BOOLEAN to FB_SIZETYPE_UINT64 ) as RANGEVALUES = _
+		{ _
+			/' FB_SIZETYPE_BOOLEAN '/ ( -1, 0, 0 ),  _
+			/' FB_SIZETYPE_INT8    '/ ( -128, 127, 127 ), _
+			/' FB_SIZETYPE_UINT8   '/ ( 0, 127, 255 ), _
+			/' FB_SIZETYPE_INT16   '/ ( -32768, 32767, 32767 ), _
+			/' FB_SIZETYPE_UINT16  '/ ( 0, 32767, 65535 ), _
+			/' FB_SIZETYPE_INT32   '/ ( -2147483648ll, 2147483647ll, 2147483647ull ), _
+			/' FB_SIZETYPE_UINT32  '/ ( 0, 2147483647ll, 4294967295ull ),  _
+			/' FB_SIZETYPE_INT64   '/ ( -9223372036854775808ll, 9223372036854775807ll, 9223372036854775807ull ), _
```

Comment:
Although `-9223372036854775808ll` seems to work "as expected" here, it may still be better to write it as `-9223372036854775807ll - 1` to avoid the constant overflow.

Also, how about writing these as hex to make typos easier to spot?

Reply:
That makes sense.  I changed the table values to hex notation.  I also considered moving the table to symb-data.bas; there are a number of places in the compiler where these range limit values are used.  Or, maybe add a "fbc.bi" or "fb-limits.bi", either in the src/compiler directory, or in the ./inc directory and document in wiki.  Similiar to limits.h.